src/common/streamreader.h

```diff
@@ -47,6 +48,21 @@ class StreamReader {
     std::string getCString();
     std::string getString(int len);
 
+    void read(char &obj); // alias for getByte
```

Comment:
What's the point of so many different "read" interfaces? I'd much rather keep the original functions and add a generic `T get()` function for delta compression. Not a fan of the `>>` operator in particular.

Reply:
My plan was to add functions I need for this PR without replacing old functions to minimize conflicts. And potentially update/remove old functions in a separate PR.

```c++
    template <class T>
    void read(T &obj) {
        _stream->read(reinterpret_cast<char*>(&obj), sizeof(T));
        swapBytesIfNotSystemEndianess(obj, _endianess);
    }
```

is necessary here for all integral types (so I don't have to explicitly instantiate all of them in .cpp file).
While `void read(char &obj);` may be redundant here, it bypasses byte-swapping loop from the generic function.

What's wrong with `>>` operator?